src/comp/comp.py

```diff
@@ -1,71 +1,81 @@
-    Human("Daphne", 30),
-    Human("Eve", 26),
-    Human("Frank", 18),
-    Human("Glenn", 42),
-    Human("Harrison", 12),
-    Human("Igon", 41),
-    Human("David", 31),
-]
-
-# Write a list comprehension that creates a list of names of everyone
-# whose name starts with 'D':
-print("Starts with D:")
-a = []
-print(a)
-
-# Write a list comprehension that creates a list of names of everyone
-# whose name ends in "e".
-print("Ends with e:")
-b = []
-print(b)
-
-# Write a list comprehension that creates a list of names of everyone
-# whose name starts with any letter between 'C' and 'G' inclusive.
-print("Starts between C and G, inclusive:")
-c = []
-print(c)
-
-# Write a list comprehension that creates a list of all the ages plus 10.
-print("Ages plus 10:")
-d = []
-print(d)
-
-# Write a list comprehension that creates a list of strings which are the name
-# joined to the age with a hyphen, for example "David-31", for all humans.
-print("Name hyphen age:")
-e = []
-print(e)
-
-# Write a list comprehension that creates a list of tuples containing name and
-# age, for example ("David", 31), for everyone between the ages of 27 and 32,
-# inclusive.
-print("Names and ages between 27 and 32:")
-f = []
-print(f)
-
-# Write a list comprehension that creates a list of new Humans like the old
-# list, except with all the names uppercase and the ages with 5 added to them.
-# The "humans" list should be unmodified.
-print("All names uppercase:")
-g = []
-print(g)
-
-# Write a list comprehension that contains the square root of all the ages.
-print("Square root of ages:")
-import math
-h = []
-print(h)
+import re
+# The following list comprehension exercises will make use of the 
+# defined Human class. 
+class Human:
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+
+    def __repr__(self):
+        return f"<Human: {self.name}, {self.age}>"
+
+humans = [
+    Human("Alice", 29),
+    Human("Bob", 32),
+    Human("Charlie", 37),
+    Human("Daphne", 30),
+    Human("Eve", 26),
+    Human("Frank", 18),
+    Human("Glenn", 42),
+    Human("Harrison", 12),
+    Human("Igon", 41),
+    Human("David", 31),
+]
+
+# Write a list comprehension that creates a list of names of everyone
+# whose name starts with 'D':
+print("Starts with D:")
+a = [i.name for i in humans if i.name[0] == "D"]
+# for i in humans:
+#     if i.name[0] == "D":
+#         a.append(i.name)
+print(a)
+
+# Write a list comprehension that creates a list of names of everyone
+# whose name ends in "e".
+print("Ends with e:")
+b = [i.name for i in humans if i.name[len(i.name) - 1] == "e"]
+# for i in humans:
+#     num = len(i.name) - 1
+#     if i.name[num] == "e":
+#         b.append(i.name)
+print(b)
+
+# Write a list comprehension that creates a list of names of everyone
+# whose name starts with any letter between 'C' and 'G' inclusive.
+print("Starts between C and G, inclusive:")
+r = "[C-G]"
+c = [i.name for i in humans if re.findall(r, i.name[0])]

```

Comment:
I really like the use of the regex here! 

There's another solution you may or may not have tried that's maybe little more appropriate for the small range of letters you're comparing to. But, for finding more complex matches making use of regex seems like the right way to go, so I think it's 100% worthwhile that you investigated a regex solution.

An alternate solution would be taking advantage of the fact that strings are iterable. This lets you make use of the `in` keyword.

```
if i.name in "CDEFG"
```

But again, I really like the regex solution. I'm actually noting this for myself. Nice job!